From 52379ebd0e5d224d665541c5e3e422695fdb0d60 Mon Sep 17 00:00:00 2001
From: Girish Sharma <girish.sharma@einfochips.com>
Date: Mon, 18 Jan 2016 21:06:16 +0530
Subject: [PATCH 1/2] Ethernet: Provide Android layer perm. to eth0

These changes are required to expose /dev/eth0 to above Android layer to
have full access to Android applications

Change-Id: I2eaca424773747f645da58f4bd46ffd1b186c6fb
Signed-off-by: eInfochips Team <ei-qc8064@einfochips.com>
---
 common.mk                |  3 ++-
 rootdir/etc/init.qcom.rc | 22 ++++++++++++++++++++++
 2 files changed, 24 insertions(+), 1 deletion(-)

diff --git a/common.mk b/common.mk
index 349e102..940dc35 100644
--- a/common.mk
+++ b/common.mk
@@ -571,7 +571,8 @@ PRODUCT_COPY_FILES := \
     frameworks/native/data/etc/android.hardware.sensor.gyroscope.xml:system/etc/permissions/android.hardware.sensor.gyroscope.xml \
     frameworks/native/data/etc/android.hardware.usb.accessory.xml:system/etc/permissions/android.hardware.usb.accessory.xml \
     frameworks/native/data/etc/android.hardware.usb.host.xml:system/etc/permissions/android.hardware.usb.host.xml \
-    frameworks/native/data/etc/android.hardware.bluetooth_le.xml:system/etc/permissions/android.hardware.bluetooth_le.xml
+    frameworks/native/data/etc/android.hardware.bluetooth_le.xml:system/etc/permissions/android.hardware.bluetooth_le.xml \
+    frameworks/native/data/etc/android.hardware.ethernet.xml:system/etc/permissions/android.hardware.ethernet.xml
 #    packages/wallpapers/LivePicker/android.software.live_wallpaper.xml:system/etc/permissions/android.software.live_wallpaper.xml \
 
 PRODUCT_COPY_FILES += \
diff --git a/rootdir/etc/init.qcom.rc b/rootdir/etc/init.qcom.rc
index bde6926..5db19cd 100755
--- a/rootdir/etc/init.qcom.rc
+++ b/rootdir/etc/init.qcom.rc
@@ -62,6 +62,10 @@ service modem-sh /system/bin/sh /system/etc/init.qcom.modem_links.sh
     oneshot
 
 on boot
+    # create dhcpcd dir
+    mkdir /data/misc/dhcp 0770 dhcp dhcp
+    chmod 0770 /data/misc/dhcp
+    chown dhcp dhcp /data/misc/dhcp
     chown bluetooth bluetooth /sys/module/bluetooth_power/parameters/power
     chown bluetooth bluetooth /sys/class/rfkill/rfkill0/type
     chown bluetooth bluetooth /sys/class/rfkill/rfkill0/state
@@ -458,6 +462,24 @@ service wpa_supplicant /system/bin/wpa_supplicant \
     disabled
     oneshot
 
+service dhcpcd_eth0 /system/bin/dhcpcd -aABDKL
+    class main
+   disabled
+    oneshot
+
+service iprenew_eth0 /system/bin/dhcpcd -n
+     class late_start
+     disabled
+     oneshot
+
+on property:sys.boot_completed=1
+    start dns_eth0
+
+service dns_eth0 /system/bin/sh /system/bin/eth0_dns.sh
+    disabled
+    oneshot
+    seclabel u:r:shell:s0
+
 service dhcpcd_wlan0 /system/bin/dhcpcd -ABKLG
     class late_start
     disabled
-- 
1.9.1

